### PR TITLE
Allow for changing cards to play during first half of the game

### PR DESCRIPTION
**Pull request by sanderlenaerts**

### CODE REVIEW

File: src/components/settings/settings.html [hunk 1]
```diff
@@ -0,0 +1,13 @@
+<ion-list>
+  <ion-list-header>Game Settings</ion-list-header>
+  <ion-item>
+      <ion-label>Maximum cards</ion-label>
+      <ion-select [disabled]="!goingUp()" [(ngModel)]="settings.maxCards" (ionChange)="changeCardsToPlay($event)">
+        <ion-option *ngFor="let card of cards" value="{{card}}">{{card}}</ion-option>
+      </ion-select>
+    </ion-item>
+  <button ion-item (click)="restart()">
```
Comment (bdhoine) [inline src/components/settings/settings.html]: Fix indentation

File: src/app/app.module.ts [hunk 2]
```diff
@@ -23,7 +25,8 @@ import { SettingsProvider } from '../providers/settings/settings';
   ],
   bootstrap: [IonicApp],
   entryComponents: [
-    MyApp
+    MyApp,
+    SettingsComponent
```
Comment (bdhoine) [inline src/app/app.module.ts]: Why did you create a component for this and not just a page and a provider?

File: src/components/settings/settings.ts [hunk 3]
```diff
@@ -0,0 +1,73 @@
+import { Round } from './../../models/round';
+import { Component } from '@angular/core';
+import { ViewController, AlertController, App, NavParams } from 'ionic-angular';
+import { RoundsProvider } from '../../providers/rounds/rounds';
+import { SettingsProvider } from '../../providers/settings/settings';
+import { GameSettings } from '../../models/gamesettings';
+
+/**
```
Comment (bdhoine) [inline src/components/settings/settings.ts]: Remove auto generated comments

File: src/components/settings/settings.ts [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+    public viewCtrl: ViewController, 
+    public roundsProvider: RoundsProvider, 
+    public alertCtrl: AlertController, 
+    public appCtrl: App,
+    public settingsProvider: SettingsProvider,
+    public navParams: NavParams
+  ) { 
+    this.cards = [];
+    this.currentRound = this.navParams.get('round');
+    console.log(this.settings, this.settingsProvider.settings)
```
Comment (bdhoine) [inline src/components/settings/settings.ts]: No console log messages

File: src/pages/new-game/new-game.ts [hunk 5]
```diff
@@ -30,23 +26,23 @@ export class NewGamePage {
...
     this.newPlayer = '';
     this.players = [];
-    this.settings = {
-      maxCards: 7,
-      cardsToPlay: GameType.ALL
-    };
+  }
+
+  get settings() {
+    console.log(this.settingsProvider.settings)
```
Comment (bdhoine) [inline src/pages/new-game/new-game.ts]: No console logs

File: src/pages/new-game/new-game.ts [hunk 6]
```diff
@@ -30,23 +26,23 @@ export class NewGamePage {
...
-    });
-    this.settingsProvider.loadSettings(this.settings).then((settings: GameSettings) => {
-      this.settings = settings;
+      this.updateCardsSelect();
     });
   }
 
   ionViewWillLeave() {
     this.playersProvider.savePlayers(this.players);
+    console.log("Leave")
```
Comment (bdhoine) [inline src/pages/new-game/new-game.ts]: No console logs

File: src/providers/settings/settings.ts [hunk 7]
```diff
@@ -36,6 +86,8 @@ export class SettingsProvider {
   }
 
   saveSettings(settings:any) {
+    console.log("Saving settings: ", settings)
```
Comment (bdhoine) [inline src/providers/settings/settings.ts]: No console logs

File: src/theme/variables.scss [hunk 8]
```diff
@@ -38,7 +38,8 @@ $colors: (
   secondary:  #32db64,
   danger:     #f53d3d,
   light:      #f4f4f4,
-  dark:       #222
+  dark:       #222,
+  grey: rgba(0, 0, 0, 0.35)
```
Comment (bdhoine) [inline src/theme/variables.scss]: Choose one color definition

Comment (sanderlenaerts) [reply src/theme/variables.scss]: The ionic build does not support 4/8 digit hex code to include transparency yet

File: src/components/settings/settings.scss [hunk 9]
```diff
@@ -0,0 +1,3 @@
+settings {
```
Comment (bdhoine) [inline src/components/settings/settings.scss]: no need to commit a scss file without any styling

File: src/providers/rounds/rounds.ts [hunk 10]
```diff
@@ -1,15 +1,28 @@
...
 import { Injectable } from '@angular/core';
 import { Storage } from '@ionic/storage';
 import { AlertController, NavController } from 'ionic-angular';
 import { GameType } from '../../models/gametype';
+import { GameSettings } from '../../models/gamesettings';
 
 @Injectable()
 export class RoundsProvider {
 
+  private _rounds: Round[];
```
Comment (bdhoine) [inline src/providers/rounds/rounds.ts]: why the _ prefix, it's nowhere used in other files

File: src/theme/variables.scss [hunk 11]
```diff
@@ -38,7 +38,8 @@ $colors: (
   secondary:  #32db64,
   danger:     #f53d3d,
   light:      #f4f4f4,
-  dark:       #222
+  dark:       #222,
+  translucent: rgba(0, 0, 0, 0.35)
```
Comment (bdhoine) [inline src/theme/variables.scss]: translucent is not really a color :) also we should use only one color definition

File: src/providers/settings/settings.ts [hunk 12]
```diff
@@ -22,11 +66,17 @@ export class SettingsProvider {
     return mergedSettings as GameSettings;
   }
 
+  getStoredSettings = async (): Promise<GameSettings> => {
+    return this.storage.get('settings').then((settings: string) => {
+      return JSON.parse(settings) as GameSettings
```
Comment (bdhoine) [inline src/providers/settings/settings.ts]: will this return a Promise?

File: src/providers/settings/settings.ts [hunk 13]
```diff
@@ -22,11 +66,17 @@ export class SettingsProvider {
     return mergedSettings as GameSettings;
   }
 
+  getStoredSettings = async (): Promise<GameSettings> => {
```
Comment (bdhoine) [inline src/providers/settings/settings.ts]: I don't like the mixing of async and other async code execution mechanisms, use consistent through codebase

File: src/providers/settings/settings.ts [hunk 14]
```diff
@@ -2,11 +2,55 @@ import { Injectable } from '@angular/core';
...
+  private roundCardsToPlay(maxCards: number) {
+    if (!this.isValidCardsAmount(this.settings)) {
+      maxCards--;
+    }
+    return maxCards;
+  }
+
+  public isValidCardsAmount(settings: GameSettings) {
+    return (settings.cardsToPlay === GameType.ODD && isOdd(settings.maxCards))
+      || (settings.cardsToPlay === GameType.EVEN && !isOdd(settings.maxCards) || settings.cardsToPlay === GameType.ALL)
```
Comment (bdhoine) [inline src/providers/settings/settings.ts]: I know this is not new code but please put newline at second or also

File: src/providers/rounds/rounds.ts [hunk 15]
```diff
@@ -144,7 +166,15 @@ export class RoundsProvider {
       rounds.push(this.generateRound(cards, players, dealer));
       dealer = this.nextPlayer(players, dealer);
     }
+    return rounds;
+  }
+
+  generateRounds(settings:any, players:string[], dealer:string):any {
+    const rounds = this.createRounds(settings, players, dealer);
```
Comment (bdhoine) [inline src/providers/rounds/rounds.ts]: useless assignment, do this in one line

**Pull request closed without merging**